Assets/Scripts/ECS/Movement/Pathfinding/RemoveOldUnreachableSystem.cs

```diff
@@ -0,0 +1,33 @@
+using Ecosystem.ECS.Movement.Pathfinding;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Jobs;
+using Unity.Mathematics;
+
+/// <summary>
+/// Removes UnreachablePosition elements from an entity's buffer based on its timestamp.
+/// </summary>
+public class RemoveOldUnreachableSystem : SystemBase
+{
+    private const double REMOVE_TIME = 60.0; // seconds
+
+    protected override void OnUpdate()
+    {
+        double now = Time.ElapsedTime;
+
+        Entities
+        .WithoutBurst()
+        .ForEach((Entity entity,
+            ref DynamicBuffer<UnreachablePosition> unreachablePositions) =>
+        {
+            for (int i = 0; i < unreachablePositions.Length; i++)
+            {
+                double difference = now - unreachablePositions[i].Timestamp;
+                if (difference > REMOVE_TIME)
+                {
+                    unreachablePositions.RemoveAt(i);
+                }
+            }
```

Comment:
```suggestion
            while (unreachablePositions.Length > 0)
            {
                double difference = now - unreachablePositions[0].Timestamp;
                if (difference < REMOVE_TIME) break;

                unreachablePositions.RemoveAt(0);
            }
```
Det blir lätt fel om man ändrar på listan som man itererar över. Om man tar bort på position i och sedan increment:ar i så skippar man ett element.

Eftersom alla element ligger i ordning (äldst först) behöver vi inte gå igenom alla. Vi kan pop:a början av listan tills vi når ett element som inte är tillräckligt gammalt och sedan avsluta loopen.

Reply:
Ja, jag tänkte på det felet också. Det är en bättre lösning.